src/frameworks/flutter/index.ts

```diff
@@ -0,0 +1,42 @@
+import { sync as execSync } from "cross-spawn";
+import { copy, pathExists } from "fs-extra";
+import { mkdir } from "fs/promises";
+import { join } from "path";
+import { BuildResult, Discovery, FrameworkType, SupportLevel } from "..";
+import * as yaml from "js-yaml";
+import { readFile } from "fs/promises";
+import { FirebaseError } from "../../error";
+
+export const name = "Flutter";
+export const type = FrameworkType.Framework;
+export const support = SupportLevel.Experimental;
+
+export async function discover(dir: string): Promise<Discovery | undefined> {
+  if (!(await pathExists(join(dir, "pubspec.yaml")))) return;
+  if (!(await pathExists(join(dir, "web")))) return;
+  const pubSpecBuffer = await readFile(join(dir, "pubspec.yaml"));
+  const pubSpec = yaml.load(pubSpecBuffer.toString());
+  const usingFlutter = pubSpec.dependencies?.flutter;
+  if (!usingFlutter) return;
+  return { mayWantBackend: false, publicDirectory: join(dir, "web") };
+}
+
+function checkForFlutterCLI() {
+  const flutter = execSync(`flutter --version`, { stdio: "ignore" });
+  if (flutter.status) throw new FirebaseError("Flutter CLI not found.");
+}
+
+export function build(cwd: string): Promise<BuildResult> {
+  checkForFlutterCLI();
+  const build = execSync(`flutter build web --release`, {
```

Comment:
```suggestion
  const build = execSync(`flutter build web`, {
```

`--release` is the default

Reply:
How is customization handled here? Folks might want to pass a number of things to the CLI...

Reply:
Fair q. 

Short term, we have a target environment variable landing soon e.g, `FIREBASE_HOSTING_BUILD_TARGET=profile`
Long term, one will be able to override the build command.

Mid term... what are the most common? 

I could see respecting a `FIREBASE_HOSTING_BUILD_ARGS="..."` env variable